src/azure-cli/azure/cli/command_modules/storage/azcopy/util.py

```diff
@@ -36,6 +36,8 @@ def __init__(self, creds=None):
                 os.makedirs(install_dir)
             base_url = 'https://azcopyvnext.azureedge.net/release20190423/azcopy_{}_amd64_10.1.0.{}'
             if self.system == 'Windows':
+                if platform.machine() != 'AMD64':
```

Comment:
This would not work for OSX & Linux

Reply:
Right! I will deprecate this change and declare in help message. But when 32-bit is ready, we have to consider how to detect OS architecture again. From my side, platform.machine() can return values related to architecture. It seems that it will return AMD64 for 64-bit Windows and x86-64 for 64-bit Linux OS. For MacOS, I have no idea and cannot try anywhere.

Reply:
One more consideration is that it is a little strange to warning in help message. I think it will be a better way to detect the OS architecture and warning when not matched.